src/elements/common/item/__tests__/IconCell.test.js

```diff
@@ -0,0 +1,94 @@
+
+    describe('render()', () => {
+        test('should render default file icon', () => {
+            const rowData = { type: undefined };
+            const wrapper = getWrapper({ rowData });
+
+            expect(wrapper.name()).toEqual('FileIcon');
+            expect(wrapper.prop('extension')).toEqual(undefined);
+            expect(wrapper.prop('title')).toEqual('File');
+        });
+
+        [
+            // personalFolder
+            {
+                rowData: {
+                    type: 'folder',
+                    has_collaborations: false,
+                    is_externally_owned: false,
+                },
+                title: 'Personal Folder',
+            },
+            // collabFolder
+            {
+                rowData: {
+                    type: 'folder',
+                    has_collaborations: true,
+                    is_externally_owned: false,
+                },
+                title: 'Collaborated Folder',
+            },
+            // externalCollabFolder
+            {
+                rowData: {
+                    type: 'folder',
+                    has_collaborations: true,
+                    is_externally_owned: true,
+                },
+                title: 'Collaborated Folder',
+            },
+            // externalFolder
+            {
+                rowData: {
+                    type: 'folder',
+                    has_collaborations: false,
+                    is_externally_owned: true,
+                },
+                title: 'External Folder',
+            },
+        ].forEach(({ rowData, title }) => {
+            test('should render correct folder icon', () => {
+                const wrapper = getWrapper({ rowData });
+
+                expect(wrapper.name()).toEqual('FolderIcon');
+                expect(wrapper.prop('isCollab')).toEqual(rowData.has_collaborations);
+                expect(wrapper.prop('isExternal')).toEqual(rowData.is_externally_owned);
+                expect(wrapper.prop('title')).toEqual(title);
+
+                expect(wrapper).toMatchSnapshot();
+            });
+        });
+
+        test('should render correct file icon', () => {
+            const extension = 'boxnote';
+            const rowData = { type: 'file', extension };
+            const wrapper = getWrapper({ rowData });
+
+            expect(wrapper.name()).toEqual('FileIcon');
+            expect(wrapper.prop('extension')).toEqual(extension);
+            expect(wrapper.prop('title')).toEqual('File');
+
+            expect(wrapper).toMatchSnapshot();
+        });
+
+        test('should render correct bookmark icon', () => {
+            const rowData = { type: 'web_link' };
+            const wrapper = getWrapper({ rowData });
+
+            expect(wrapper.name()).toEqual('BookmarkIcon');
+            expect(wrapper.prop('title')).toEqual('Bookmark');
+
+            expect(wrapper).toMatchSnapshot();
```

Comment:
Do you think these snapshots are necessary?

Reply:
They are probably redundant